### PR TITLE
feat(schedule):support common quartz schedule job

### PR DESCRIPTION
**Pull request by PeachThinking**
#### What type of PR is this?
module-quartz

#### What this PR does / why we need it:
support common quartz schedule job which is not related to ODC scheduled tasks.
common quartz job will use `commonScheduler` to schedule job.
add `private Class<? extends Job> jobClass` in `CreateQuartzJobParam` and `ChangeQuartJobParam` to determine whether to use `commonScheduler` or `defaultScheduler`.

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by guowl3 — APPROVED** [review]
lgtm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobService.java [hunk 1]
```diff
@@ -59,11 +59,15 @@ public class QuartzJobService {
...
-    private Scheduler scheduler;
+    private Scheduler defaultScheduler;
+    @Autowired(required = false)
+    @Qualifier(value = ("commonScheduler"))
+    private Scheduler commonScheduler;
 
     public void changeQuartzJob(ChangeQuartJobParam req) {
 
         JobKey jobKey = QuartzKeyGenerator.generateJobKey(req.getJobName(), req.getJobGroup());
+        Scheduler scheduler = QuartzJob.class.isAssignableFrom(req.getJobClazz()) ? defaultScheduler : commonScheduler;
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobService.java:70]: clazz may be a typo?

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobService.java [hunk 2]
```diff
@@ -59,11 +59,15 @@ public class QuartzJobService {
 
     @Autowired(required = false)
     @Qualifier(value = ("defaultScheduler"))
-    private Scheduler scheduler;
+    private Scheduler defaultScheduler;
+    @Autowired(required = false)
+    @Qualifier(value = ("commonScheduler"))
+    private Scheduler commonScheduler;
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobService.java:65]: this modification exits a issue is that changeQuartzJob can retrieve job class name, but pause or other operation can't know what type of this job is, user should pass scheduler to method. as user has know what scheduler is, why not remote auto judge code and let scheduler as a input for function call

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobService.java (hunk 1 above)
Comment (PeachThinking) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobService.java:70]: This is my spelling mistake, I have corrected it

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobService.java (hunk 2 above)
Comment (PeachThinking) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobService.java:65]: i create a new QuartzJobServiceProxy to solve this problem

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/config/QuartzConfiguration.java [hunk 3]
```diff
@@ -50,6 +50,8 @@ public class QuartzConfiguration {
     private Long maxConcurrentTaskCount;
 
     private final String defaultSchedulerName = "ODC-SCHEDULER";
+    private final String commonSchedulerName = "ODC-COMMON-SCHEDULER";
+    private final String APPLICATION_CONTEXT_SCHEDULER_KEY_NAME = "applicationContext";
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/config/QuartzConfiguration.java:54]: static final use UPPER variable name

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobServiceProxy.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        getQuartzJobServiceImpl(key.getGroup()).triggerJob(key, triggerDataMap);
+    }
+
+    public List<? extends Trigger> getJobTriggers(JobKey jobKey) throws SchedulerException {
+        return getQuartzJobServiceImpl(jobKey.getGroup()).getJobTriggers(jobKey);
+    }
+
+    private AbstractQuartzJobService getQuartzJobServiceImpl(String group) {
+        for (ScheduleTaskType type : ScheduleTaskType.values()) {
+            if (type.name().equalsIgnoreCase(group)) {
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/QuartzJobServiceProxy.java:105]: use uppercase map to find group may be more quickly

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/AbstractQuartzJobService.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+    public void triggerJob(JobKey key, JobDataMap triggerDataMap) throws SchedulerException {
+        getScheduler().triggerJob(key, triggerDataMap);
+    }
+
+    public List<? extends Trigger> getJobTriggers(JobKey jobKey) throws SchedulerException {
+        return getScheduler().getTriggersOfJob(jobKey);
+    }
+
+    abstract protected Scheduler getScheduler();
```
Comment (LioRoger) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/AbstractQuartzJobService.java:211]: there maybe a common declare order, like pubic/private static final violate type eg.

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/config/QuartzConfiguration.java (hunk 3 above)
Comment (PeachThinking) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/config/QuartzConfiguration.java:54]: but “applicationContext” is a normal name for setApplicationContextSchedulerContextKey

File: server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/AbstractQuartzJobService.java (hunk 5 above)
Comment (PeachThinking) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/quartz/AbstractQuartzJobService.java]: 👌

**Review by LioRoger — APPROVED** [review]
lgtm

**Review by guowl3 — APPROVED** [review]
lgtm

**Pull request merged**